src/timed_string.rs

```diff
@@ -0,0 +1,78 @@
+use std::time::{Duration, Instant};
+use bytes::Bytes;
+
+pub struct TimedString {
+    expires_at: Instant,
+    cache_for:  Duration,
+    url:  String,
+    pub text: Bytes,
+    pub updater: fn (&str) -> Bytes
```

Comment:
having a function that handles updating the cache means we can stub this function out in tests to do nothing; I'm not sure if this is a weird approach or not, I don't know what the best practice is here.